x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/InTests.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.esql.expression.predicate.operator.comparison;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.esql.core.TestUtils;
+import org.elasticsearch.xpack.esql.core.expression.Literal;
+
+import java.util.Arrays;
+
+import static org.elasticsearch.xpack.esql.core.expression.Literal.NULL;
+import static org.elasticsearch.xpack.esql.core.tree.Source.EMPTY;
+
+public class InTests extends ESTestCase {
```

Comment:
Should we also remove the corresponding test class in `esql-core`, or at least comment it/mark it as deprecated?

Reply:
I didn't remove it just because there are still some core functionalities relying on `In`. Next step will be to rationalize the expression hierarchies, removing the comparison implementations in core and moving all the logic to the ESQL versions; at that point I'll also remove this test.